lib/connectors/base/simple_rules_parser.rb

```diff
@@ -11,23 +11,144 @@
 
 module Connectors
   module Base
+    class FilteringRulesValidationError < StandardError; end
+
     class SimpleRulesParser
+
+      include Core::Filtering
+
+      attr_reader :rules
+
       def initialize(rules)
-        @rules = (rules || []).map(&:with_indifferent_access).filter { |r| r[:id] != 'DEFAULT' }.sort_by { |r| r[:order] }
+        begin
+          sorted = (rules || []).map { |r| SimpleRule.new(r) }.filter { |r| r.id != 'DEFAULT' }.sort_by(&:order)
+        rescue StandardError => e
+          raise FilteringRulesValidationError, "Invalid rule: #{e.message}"
+        end
+        @rules = validate(sorted)
       end
 
       def parse
-        merge_rules(@rules.map do |rule_hash|
-          rule = Core::Filtering::SimpleRule.new(rule_hash)
-          unless rule.is_include? || rule.is_exclude?
-            raise "Unknown policy: #{rule.policy}"
```

Comment:
Did we lose the `"unknown policy"` check?

Reply:
No, the individual rule checks went into rule init.